File: mycollab-web/src/main/scala/com/mycollab/module/project/event/InvoiceEvent.scala
package com.mycollab.module.project.event

import com.mycollab.events.ApplicationEvent
import com.mycollab.module.project.domain.SimpleInvoice

/**
  * @author MyCollab Ltd
  * @since 5.2.10
  */
object InvoiceEvent {

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class NewInvoiceAdded(source: AnyRef, data: SimpleInvoice) extends ApplicationEvent(source, data) {}

  class InvoiceUpdateAdded(source: AnyRef, data: SimpleInvoice) extends ApplicationEvent(source, data) {}

  class InvoiceDelete(source: AnyRef, data: SimpleInvoice) extends ApplicationEvent(source, data) {}

  class DisplayInvoiceView(source: AnyRef, data: SimpleInvoice) extends ApplicationEvent(source, data) {}

}

File: mycollab-web/src/main/java/com/mycollab/module/project/event/InvoiceEvent.kt
package com.mycollab.module.project.event

import com.mycollab.vaadin.event.ApplicationEvent
import com.mycollab.module.project.domain.SimpleInvoice

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object InvoiceEvent {
    class GotoList(source: Any, val data: Any?) : ApplicationEvent(source)

    class NewInvoiceAdded(source: Any, data: SimpleInvoice) : ApplicationEvent(source)

    class InvoiceUpdateAdded(source: Any, data: SimpleInvoice) : ApplicationEvent(source)

    class InvoiceDelete(source: Any, data: SimpleInvoice) : ApplicationEvent(source)

    class DisplayInvoiceView(source: Any, data: SimpleInvoice) : ApplicationEvent(source)
}